### ISSUE TITLE
`max_cat_threshold` Hyperparameter

### ISSUE THREAD

**Issue by pford221** [microsoft/LightGBM#2261]
Hi, 

Is there another way to find out what is happening with `max_cat_threhsold` without trying to find it in the C++ source code? To save anyone kind enough to reply some time, I'll share my hypothesis and hopefully you can just correct me if I'm wrong?

Suppose there is a categorical variable of cardinality 10 (i.e. 10 unique levels). Also suppose that we set `max_cat_threshold = 1`. At each split opportunity, the algorithm aggregates the sum(gradients) / sum(hessians) for all the records in each of the 10 categories and then sorts the 10 categories by that ratio from lowest to highest (or highest to lowest...doesn't matter). Then because `max_cat_threshold = 1`, the algorithm only has one split point to evaluate and this split point is as close to the median (as determined by number of observations or weighted observations) as possible?

I appreciate any help on this!

**Comment by guolinke** [comment]
@pford221 yeah, you are right. max_cat_threshold is to limit the categorical split points.

**Comment by pford221** [comment]
Thank you, @guolinke. I realize that `max_cat_threshold` limits split points, but would you mind commenting on if my description of how it does it more-or-less correct? I really appreciate it.

**Comment by guolinke** [comment]
@pford221 
I think most of them are correct.
But for 

> Then because max_cat_threshold = 1, the algorithm only has one split point to evaluate and this split point is as close to the median (as determined by number of observations or weighted observations) as possible?

, I want to clarify that, the split point will be the one with the highest (or lowest) `sum(gradients) / sum(hessians) `.
